CoreUtilities/src/au/gov/asd/tac/constellation/utilities/icon/IconData.java

```diff
@@ -38,6 +38,15 @@
 
     private byte[] data = null;
 
+    /**
+     * Setter used only for testing.
+     *
+     * @param data
+     */
+    protected void setData(final byte[] data) {
+        this.data = data;
+    }
```

Comment:
is there a way your unit tests can be written such that this setter isn't needed? I tend to be ok with getters being added for unit testing but a little more reluctant to add setters if unit testing is their only use

Reply:
I can't seem to get data be another value without doing so, as data is assigned the value of a bytearray from an outputstream that seems to always be emtpy after the stream is used.

Reply:
@sol695510 see my below comments. I've added some thoughts on alternatives